main.go

```diff
@@ -392,5 +433,9 @@ func shutdownBlock(ifce *Interface) {
 	ifce.hostMap.Unlock()
 
 	l.WithField("signal", sig).Info("Goodbye")
-	os.Exit(0)
+	if cmd.Callback != nil {
+		select {
+		case cmd.Callback <- nil:
+		}
+	}
```

Comment:
You don't need the nil check here, and if you want the select to be non-blocking you need to add a `default:` case